operator-pipeline-images/operatorcert/utils.py

```diff
@@ -134,33 +134,42 @@ def __call__(
 
 
 def run_command(
-    cmd: List[str], check: bool = True, cwd: Optional[str] = None
+    cmd: List[str], check: bool = True, cwd: Optional[str] = None, retries: int = 1
 ) -> subprocess.CompletedProcess[bytes]:
     """
     Run a shell command and return its output.
 
     Args:
         cmd (List[str]): Command to run
+        check (bool): Whether to check the command exit code
+        cwd (Optional[str]): Current working directory
+        retries (int): Number of retries before failing
 
     Returns:
         CompletedProcess: Command output
     """
     LOGGER.debug("Running command: %s", cmd)
-    try:
-        output = subprocess.run(
-            cmd,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            check=check,
-            cwd=cwd,
-        )
-    except subprocess.CalledProcessError as e:
-        LOGGER.error(
-            "Error running command: \nstdout: %s\nstderr: %s",
-            e.stdout,
-            e.stderr,
-        )
-        raise e
+    for attempt in range(1, retries + 1):
+        try:
+            output = subprocess.run(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                check=check,
+                cwd=cwd,
+            )
```

Comment:
Shouldn't there be a `break` statement at the end of the try block? Or what happens if there is successful attempt before the retries are exhausted?

Reply:
Good catch! It is fixed and covered by the unit tests now.